external-crates/move/crates/move-bytecode-verifier/src/signature.rs

```diff
@@ -15,22 +15,41 @@ use move_binary_format::{
     file_format_common::VERSION_6,
     IndexKind,
 };
+use move_bytecode_verifier_meter::{Meter, Scope};
 use move_core_types::vm_status::StatusCode;
-use std::collections::{HashMap, HashSet};
+use std::{
+    cell::RefCell,
+    collections::{HashMap, HashSet},
+};
+
+use crate::ability_cache::AbilityCache;
 
-pub struct SignatureChecker<'a> {
-    module: &'a CompiledModule,
+pub struct SignatureChecker<'env, 'a, 'b, M: Meter + ?Sized> {
+    module: &'env CompiledModule,
+    module_ability_cache: RefCell<&'a mut AbilityCache<'env>>,
+    meter: RefCell<&'b mut M>,
```

Comment:
Is there a particular reason for these `RefCell`s? Would be nice if they weren't needed.

Reply:
laziness, fixed